<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phylomapr">
<title>Adding phylomaps to phylomapr • phylomapr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Adding phylomaps to phylomapr">
<meta property="og:description" content="phylomapr">
<meta property="og:image" content="https://lotharukpongjs.github.io/phylomapr/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phylomapr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Adding_phylomaps.html">Adding phylomaps to phylomapr</a>
    <a class="dropdown-item" href="../articles/GeneIDs.html">Gene names in different databases</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Adding phylomaps to phylomapr</h1>
            
            <h4 data-toc-skip class="date">2023-10-11</h4>
      
      
      <div class="d-none name"><code>Adding_phylomaps.Rmd</code></div>
    </div>

    
    
<p>This package aims to provide easy access to pre-computed phylomaps.
But the tree of life is huge (there are at least <a href="https://www.nature.com/articles/news.2011.498" class="external-link">millions of
eukaryotic species</a>) and so the production of phylomaps, be it for
evolutionary transcriptomics or other
genomics/transcriptomics/proteomics studies, is a joint effort. For
advanced gene age mappers who have generated their own phylomaps and
would like to contribute to <code>phylomapr</code>, here is a short
tutorial to do just that!</p>
<div class="section level2">
<h2 id="easy-option">Easy option<a class="anchor" aria-label="anchor" href="#easy-option"></a>
</h2>
<p>The easiest way (easy here for you, not for me) to contribute is to
just post a github issue with the link to where the phylomap is hosted
or the paper that contains the phylomap and describe the method that was
employed to obtain it (e.g. the version of gene age inference tools such
as <code>GenEra</code> and parameters of the sequence aligner)! The
maintainers of phylomapr can then add the phylomaps manually from the
link and description provided.</p>
</div>
<div class="section level2">
<h2 id="advanced-option">Advanced option<a class="anchor" aria-label="anchor" href="#advanced-option"></a>
</h2>
<p>For even more advanced phylomappers, here is how you can
contribute.</p>
<p>Note: this might change if we move the site where phylomaps are
hosted.</p>
<ol style="list-style-type: decimal">
<li>Fork the <code>phylomapr</code> repository and clone it in R (see
tutorial <a href="https://www.camplab.net/adding-code-to-an-r-package/" class="external-link">here</a>:
read until the end of <code>B. Rstudio</code> section).</li>
</ol>
<div class="section level4">
<h4 id="if-a-link-to-where-the-phylomap-is-hosted-exists">If a link to where the phylomap is hosted exists:<a class="anchor" aria-label="anchor" href="#if-a-link-to-where-the-phylomap-is-hosted-exists"></a>
</h4>
<ol start="2" style="list-style-type: decimal">
<li>Open the <code>DATASET.R</code> in the directory
<code>data-raw</code> and add the download line, e.g.</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">######### Manley et al., 2023 #########</span></span>
<span></span>
<span><span class="co"># download the Phylostratigraphic Maps from Manley et al., 2023</span></span>
<span><span class="co"># Rhizophagus irregularis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span> url      <span class="op">=</span> <span class="st">"https://zenodo.org/record/7713976/files/Rhizophagus_irregularis_DAOM197198_1432141_phyloranks.tsv"</span>,</span>
<span>               destfile <span class="op">=</span> <span class="st">"data-raw/Rhizophagus_irregularis_DAOM197198_1432141_phyloranks.tsv"</span><span class="op">)</span></span></code></pre></div>
<p>Here you would describe the dataset (in <code>#</code>) differently
and have a different download link.</p>
<ol start="3" style="list-style-type: decimal">
<li>In <code>DATASET.R</code>, create a <code>[Species].PhyloMap</code>
object from the downloaded file. There are many ways to do this and this
is just an example. Make sure that the name of the
<code>[Species].PhyloMap</code> object isn’t duplicated. If so, add some
more information e.g. the geneID convention
i.e. <code>[Species].ENSEMBL.PhyloMap</code>.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load package readxl</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Phylostratigraphic Maps</span></span>
<span><span class="co"># Rhizophagus irregularis</span></span>
<span><span class="va">Rhizophagus_irregularis.data</span> <span class="op">&lt;-</span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="st">"data-raw/Rhizophagus_irregularis_DAOM197198_1432141_phyloranks.tsv"</span><span class="op">)</span></span>
<span><span class="va">Rhizophagus_irregularis.PhyloMap</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">Rhizophagus_irregularis.data</span>,</span>
<span>    Phylostratum <span class="op">=</span> <span class="va">PS</span>,</span>
<span>    <span class="va">GeneID</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>The most important thing is that the <code>[Species].PhyloMap</code>
object has the first column with phylostratum titled
<code>Phylostratum</code> and the second column with the GeneID titled
<code>GeneID</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> phylomapr<span class="sc">::</span>Rhizophagus_irregularis.PhyloMap</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 31,217 × 2</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>   Phylostratum GeneID   </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>          <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span>    </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a> <span class="dv">1</span>            <span class="dv">8</span> g100<span class="sc">-</span>T1  </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a> <span class="dv">2</span>            <span class="dv">6</span> g1000<span class="sc">-</span>T1 </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a> <span class="dv">3</span>            <span class="dv">2</span> g10000<span class="sc">-</span>T1</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a> <span class="dv">4</span>            <span class="dv">9</span> g10001<span class="sc">-</span>T1</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a> <span class="dv">5</span>            <span class="dv">1</span> g10002<span class="sc">-</span>T1</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a> <span class="dv">6</span>            <span class="dv">1</span> g10003<span class="sc">-</span>T1</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a> <span class="dv">7</span>            <span class="dv">1</span> g10004<span class="sc">-</span>T1</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a> <span class="dv">8</span>            <span class="dv">1</span> g10005<span class="sc">-</span>T1</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a> <span class="dv">9</span>            <span class="dv">1</span> g10006<span class="sc">-</span>T1</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span>            <span class="dv">1</span> g10007<span class="sc">-</span>T1</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ 31,207 more rows</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>At the bottom of <code>DATASET.R</code>, run
<code>usethis::use_data()</code> on the <code>[Species].PhyloMap</code>
object you’ve created, e.g.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">Rhizophagus_irregularis.PhyloMap2</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>which leads to this:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>✔ Saving <span class="st">'Rhizophagus_irregularis.PhyloMap'</span> to <span class="st">'data/Rhizophagus_irregularis.PhyloMap.rda'</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>• Document your <span class="fu">data</span> (see <span class="st">'https://r-pkgs.org/data.html'</span>)</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Run the lines you’ve added in <code>DATASET.R</code>
</li>
</ol>
<p>For the next steps go to the section <a href="#documenting-the-new-phylomap"><strong>Documenting the new
phylomap</strong></a></p>
</div>
<div class="section level4">
<h4 id="if-you-want-to-load-your-phylomaps--tsv-or--csv-files-etc--manually">If you want to load your phylomaps (.tsv or .csv files etc.)
manually:<a class="anchor" aria-label="anchor" href="#if-you-want-to-load-your-phylomaps--tsv-or--csv-files-etc--manually"></a>
</h4>
<ol start="2" style="list-style-type: decimal">
<li><p>Add the raw output of gene age inference
(e.g. <code>[taxid]_gene_ages.tsv</code> from GenEra) to the directory
<code>data-raw</code>.</p></li>
<li><p>In <code>DATASET.R</code>, create a
<code>[Species].PhyloMap</code> object from the added raw file. There
are many ways to do this and this is just an example. Make sure that the
name of the <code>[Species].PhyloMap</code> object isn’t duplicated. If
so, add some more information e.g. the geneID convention
i.e. <code>[Species].ENSEMBL.PhyloMap</code>.</p></li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">######### Strongylocentrotus purpuratus GenEra test #########</span></span>
<span></span>
<span><span class="va">Strongylocentrotus_purpuratus.data</span> <span class="op">&lt;-</span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="st">"data-raw/7668_gene_ages.tsv"</span><span class="op">)</span></span>
<span><span class="va">Strongylocentrotus_purpuratus.PhyloMap</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">Strongylocentrotus_purpuratus.data</span>,</span>
<span>    Phylostratum <span class="op">=</span> <span class="va">rank</span>,</span>
<span>    GeneID <span class="op">=</span> <span class="va">`#gene`</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>The most important thing is that the <code>[Species].PhyloMap</code>
object has the first column with phylostratum titled
<code>Phylostratum</code> and the second column with the GeneID titled
<code>GeneID</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> phylomapr<span class="sc">::</span>Strongylocentrotus_purpuratus.PhyloMap</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 38,475 × 2</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>   Phylostratum GeneID        </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>          <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span>         </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a> <span class="dv">1</span>            <span class="dv">1</span> NP_001001474<span class="fl">.1</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a> <span class="dv">2</span>            <span class="dv">1</span> NP_001001475<span class="fl">.1</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a> <span class="dv">3</span>            <span class="dv">1</span> NP_001001476<span class="fl">.1</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a> <span class="dv">4</span>            <span class="dv">2</span> NP_001001477<span class="fl">.1</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a> <span class="dv">5</span>            <span class="dv">1</span> NP_001001478<span class="fl">.1</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a> <span class="dv">6</span>            <span class="dv">4</span> NP_001001768<span class="fl">.2</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a> <span class="dv">7</span>            <span class="dv">1</span> NP_001001906<span class="fl">.1</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a> <span class="dv">8</span>            <span class="dv">2</span> NP_001003798<span class="fl">.1</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a> <span class="dv">9</span>            <span class="dv">1</span> NP_001005725<span class="fl">.1</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span>            <span class="dv">1</span> NP_001008790<span class="fl">.1</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ 38,465 more rows</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>At the bottom of <code>DATASET.R</code>, run
<code>usethis::use_data</code> on the <code>[Species].PhyloMap</code>
object you’ve created, e.g.</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_data</span><span class="op">(</span><span class="va">Strongylocentrotus_purpuratus.PhyloMap</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>✔ Saving <span class="st">'Strongylocentrotus_purpuratus.PhyloMap'</span> to <span class="st">'data/Strongylocentrotus_purpuratus.PhyloMap.rda'</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>• Document your <span class="fu">data</span> (see <span class="st">'https://r-pkgs.org/data.html'</span>)</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Run the lines you’ve added in <code>DATASET.R</code>
</li>
</ol>
<p>For the next steps go to the section <a href="#documenting-the-new-phylomap"><strong>Documenting the new
phylomap</strong></a></p>
</div>
<div class="section level4">
<h4 id="documenting-the-new-phylomap">Documenting the new phylomap<a class="anchor" aria-label="anchor" href="#documenting-the-new-phylomap"></a>
</h4>
<ol start="6" style="list-style-type: decimal">
<li>Add the description for your new phylomapr using
<code>usethis::use_r()</code>,
i.e. <code>usethis::use_r("[Species].R")</code> (note: the format used
in <code>phylomapr</code> doesn’t contain the <code>.PhyloMap</code>
suffix). For example, if I want to add the description for
<em>Strongylocentrotus purpuratus</em>
</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">use_r</span><span class="op">(</span><span class="st">"Strongylocentrotus_purpuratus.R"</span><span class="op">)</span></span></code></pre></div>
<p>The description file contains how gene ages are inferred, any notes
on the parameters, the number of rows in the data frame, the number of
variables, the source (under <code>@source</code>) and finally
<code>"[Species].PhyloMap"</code>.</p>
<pre><code><span><span class="co">#' Phylomap of Strongylocentrotus purpuratus</span></span>
<span><span class="co">#'</span></span>
<span><span class="co">#' Gene ages inferred using [GenEra](https://github.com/josuebarrera/GenEra) on reference protein sequences from NCBI protein coding genes (https://www.ncbi.nlm.nih.gov/datasets/gene/taxon/7668/?gene-type=Protein-coding).</span></span>
<span><span class="co">#' Note: [DIAMOND](https://github.com/bbuchfink/diamond) was run using the `ultra-sensitive mode`.</span></span>
<span><span class="co">#'</span></span>
<span><span class="co">#' @format A tibble with 38,475 rows and 2 variables:</span></span>
<span><span class="co">#' \describe{</span></span>
<span><span class="co">#'   \item{Phylostratum}{dbl Phylostratum (or gene age) assignment}</span></span>
<span><span class="co">#'   \item{GeneID}{proteinID annotation from NCBI}</span></span>
<span><span class="co">#' }</span></span>
<span><span class="co">#' @source</span></span>
<span><span class="co">#' Bermejo (2023) _Unpublished_</span></span>
<span><span class="co">#' \url{https://github.com/LotharukpongJS/phylomapr}</span></span>
<span><span class="st">"Strongylocentrotus_purpuratus.PhyloMap"</span></span></code></pre>
<ol start="7" style="list-style-type: decimal">
<li><p>Document your new phylomap using
<code>devtools::document()</code> or
<code>roxygen2::roxygenise()</code>. Tada! The phylomap is documented
such that users can run <code>?[Species].PhyloMap</code> and obtain the
source and other information about the phylomap! Optionally, run
<code>devtools::check()</code> to ensure that the package is looking
good after the changes.</p></li>
<li><p>Under the <code>Git</code> tab in Rstudio (where the other tabs
include Environment, History, Connection, Buld, etc.), stage the changed
files, commit the changes (with description) and push your changes to
your fork. The staging, commit and push can alternatively be done on the
command line. Then in GitHub, make a pull request.</p></li>
</ol>
<p>Done!</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jaruwatana Sodai Lotharukpong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
