<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phylomapr">
<title>Gene names in different databases • phylomapr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene names in different databases">
<meta property="og:description" content="phylomapr">
<meta property="og:image" content="https://lotharukpongjs.github.io/phylomapr/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phylomapr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GeneIDs.html">Gene names in different databases</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Gene names in different databases</h1>
            
            <h4 data-toc-skip class="date">2023-09-26</h4>
      
      
      <div class="d-none name"><code>GeneIDs.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h2>
<p>The GeneID labels differ across databases. This is a potential
problem when gene age inference has been performed from protein
sequences obtained from one database (e.g. UniProtKB) and the RNA-seq
quantification has been performed on genes from another database
(e.g. ENSEMBL).</p>
</div>
<div class="section level2">
<h2 id="solution">Solution<a class="anchor" aria-label="anchor" href="#solution"></a>
</h2>
<p>In <code>phylomapr</code>, we try to mitigate this issue using the
helper function <code><a href="../reference/convertID.html">convertID()</a></code>, which essentially wraps around
the function <code>biomart()</code> in the <a href="https://docs.ropensci.org/biomartr/" class="external-link"><code>biomartr</code></a>
package. In this article, we give an example of how <em>C. elegans</em>
phylomaps can be converted from the uniprot format to the wormbase
format.</p>
<p>Typically, the gene IDs can be converted to the ENSEMBLE gene IDs,
whose <code>biomartr</code> attributes are <code>ensembl_gene_id</code>
and <code>ensembl_peptide_id</code> by default.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># default run of convertID</span></span>
<span><span class="va">Caenorhabditis_elegans.PhyloMap.ENSEMBL</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">phylomapr</span><span class="fu">::</span><span class="fu"><a href="../reference/convertID.html">convertID</a></span><span class="op">(</span></span>
<span>    phylomap <span class="op">=</span> <span class="fu">phylomapr</span><span class="fu">::</span><span class="va"><a href="../reference/Caenorhabditis_elegans.PhyloMap.html">Caenorhabditis_elegans.PhyloMap</a></span>,</span>
<span>    mart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span>    dataset <span class="op">=</span> <span class="st">"celegans_gene_ensembl"</span>,</span>
<span>    filters <span class="op">=</span> <span class="st">"uniprot_gn_id"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>If you run this, the gene IDs are named as
e.g. <code>WBGene00000001</code>. However, gene IDs in other RNA-seq
experiments employ other gene names. In the case of <em>C. elegans</em>
this may be wormbase IDs (e.g. <code>2L52.1a</code>). How can we convert
the phylomaps to accommodate this?</p>
<p>First, the <code>organismFilters</code> function in
<code>biomartr</code> can be used to obtain the attributes. Here, we ask
for the topic <code>worm</code> because we are interested in wormbase
IDs</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c_elegans_BM.filt</span> <span class="op">&lt;-</span> <span class="fu">organismFilters</span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"Caenorhabditis elegans"</span>, topic <span class="op">=</span> <span class="st">"worm"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c_elegans_BM.filt</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 72 × 4</span></span></span>
<span><span class="co">#&gt;    name                     description                                             dataset               mart                </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> with_wormbase_cds        With WormBase CDS ID(s)                                 celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> with_wormbase_gene       With WormBase Gene ID(s)                                celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> with_wormbase_gseqname   With WormBase Gene Sequence-name ID(s)                  celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> with_wormbase_locus      With WormBase Locus ID(s)                               celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> with_wormbase_transcript With WormBase Transcript ID(s)                          celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> with_wormpep_id          With Wormpep ID ID(s)                                   celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> wormbase_cds             WormBase CDS ID(s) [e.g. 2L52.1a]                       celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> wormbase_gene            WormBase Gene ID(s) [e.g. WBGene00000001]               celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> wormbase_gseqname        WormBase Gene Sequence-name ID(s) [e.g. WBGene00000001] celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> wormbase_locus           WormBase Locus ID(s) [e.g. WBGene00000001]              celegans_gene_ensembl ENSEMBL_MART_ENSEMBL</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 62 more rows</span></span></span></code></pre>
<p>After scrolling through the results, we observe that the attribute
<code>wormbase_cds</code> should work as the IDs are formatted in the
<code>2L52.1a.1</code>-form.</p>
<p>Next, we convert the GeneID labels from uniprot to wormbase using
<code>"wormbase_cds"</code> as the attribute.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># here we use the dataset phylomapr::Caenorhabditis_elegans.PhyloMap</span></span>
<span><span class="va">Celegans.PhyloMap</span> <span class="op">&lt;-</span> <span class="fu">phylomapr</span><span class="fu">::</span><span class="va"><a href="../reference/Caenorhabditis_elegans.PhyloMap.html">Caenorhabditis_elegans.PhyloMap</a></span></span>
<span></span>
<span><span class="va">Celegans.PhyloMap.wormbase</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">phylomapr</span><span class="fu">::</span><span class="fu"><a href="../reference/convertID.html">convertID</a></span><span class="op">(</span></span>
<span>    phylomap <span class="op">=</span> <span class="va">Celegans.PhyloMap</span>,</span>
<span>    mart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span>    dataset <span class="op">=</span> <span class="st">"celegans_gene_ensembl"</span>,</span>
<span>    attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"wormbase_cds"</span><span class="op">)</span>,</span>
<span>    filters    <span class="op">=</span> <span class="st">"uniprot_gn_id"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; Starting Gene ID conversion...</span></span></code></pre>
<pre><code><span><span class="co">#&gt; Starting BioMart query ...</span></span></code></pre>
<pre><code><span><span class="co">#&gt; Warning: Ensembl will soon enforce the use of https.</span></span>
<span><span class="co">#&gt; Ensure the 'host' argument includes "https://"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> Celegans.PhyloMap</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 19,818 × 2</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>   Phylostratum GeneID                   </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>          <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span>                    </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a> <span class="dv">1</span>            <span class="dv">2</span> sp<span class="sc">|</span>A0A061ACU2<span class="sc">|</span>PIEZ1_CAEEL</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a> <span class="dv">2</span>            <span class="dv">3</span> sp<span class="sc">|</span>A0A078BQN7<span class="sc">|</span>FKH3_CAEEL </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a> <span class="dv">3</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A078BQP2<span class="sc">|</span>GCY25_CAEEL</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a> <span class="dv">4</span>           <span class="dv">10</span> sp<span class="sc">|</span>A0A0K3AUE4<span class="sc">|</span>SEA2_CAEEL </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a> <span class="dv">5</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A0K3AUJ9<span class="sc">|</span>PRDX_CAEEL </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a> <span class="dv">6</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A0K3AV08<span class="sc">|</span>MLK1_CAEEL </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a> <span class="dv">7</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A0K3AWM6<span class="sc">|</span>MOM5_CAEEL </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a> <span class="dv">8</span>            <span class="dv">3</span> sp<span class="sc">|</span>A0A0K3AXH1<span class="sc">|</span>ARID1_CAEEL</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a> <span class="dv">9</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A131MBU3<span class="sc">|</span>IRG7_CAEEL </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span>            <span class="dv">1</span> sp<span class="sc">|</span>A0A131MCZ8<span class="sc">|</span>CNNM3_CAEEL</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ 19,808 more rows</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span> Celegans.PhyloMap.wormbase</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 16,319 × 2</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>   Phylostratum GeneID  </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>          <span class="sc">&lt;</span>dbl<span class="sc">&gt;</span> <span class="er">&lt;</span>chr<span class="sc">&gt;</span>   </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a> <span class="dv">1</span>            <span class="dv">2</span> 2RSSE<span class="fl">.1</span>a</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a> <span class="dv">2</span>            <span class="dv">2</span> 2RSSE<span class="fl">.1</span>b</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a> <span class="dv">3</span>            <span class="dv">2</span> 2RSSE<span class="fl">.1</span>c</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a> <span class="dv">4</span>           <span class="dv">14</span> 3R5<span class="fl">.1</span>a  </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a> <span class="dv">5</span>           <span class="dv">14</span> 3R5<span class="fl">.1</span>b  </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a> <span class="dv">6</span>            <span class="dv">1</span> 4R79<span class="fl">.1</span>a </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a> <span class="dv">7</span>            <span class="dv">1</span> 4R79<span class="fl">.1</span>b </span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a> <span class="dv">8</span>            <span class="dv">1</span> AC3<span class="fl">.10</span>  </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a> <span class="dv">9</span>           <span class="dv">11</span> AC3<span class="fl">.1</span>a  </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span>           <span class="dv">11</span> AC3<span class="fl">.1</span>b  </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ 16,309 more rows</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ Use `print(n = ...)` to see more rows</span></span></code></pre></div>
<p>Tada! While the conversion issue between databases are mitigated,
there are still some aspects to be improved. For example, we can see a
drop in the number of genes in the phylomap. Furthermore, a bit of
fidgeting with <code>biomartr</code> is needed. However, this should
still be faster than running gene age inference from scratch!</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jaruwatana Sodai Lotharukpong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
